src/codegen/object.rs

```diff
@@ -109,6 +111,96 @@ pub fn generate(
         try!(writeln!(w));
     }
 
+    if analysis.properties.iter().any(|property| property.construct || property.construct_only) {
```

Comment:
So it currently only generates a `Builder` if there's any construct property. But wouldn't it also make sense for some types to generate it otherwise and provide access to the non-construct, writable properties via the builder?

Another problem is that for some types it is not enough to call `g_object_new()` with their properties to create a valid instance, but the actual `foo_new()` function of the specific type has to be called.

And yet another problem I see is that object construction could fail like this if some of the properties *must* be provided during construction and the default values are not enough. This would end up as a `g_critical` and then we're in UB country. `g_object_new()` can't ever fail unfortunately, it must always return an instance... but that instance might be unusable because of this.

There are the `GInitable` and `GAsyncInitable` interfaces in `gio` for that very reason, but they're not widely used.

I think what I'm saying here is that the generation of the builders for now should probably be opt-in per type.

Reply:
Can you give examples of types where `g_object_new()` is not enough?

You're right, I forgot the properties without default values. For that, I thought I could just add the required properties in the `new()` method.

Reply:
> Can you give examples of types where g_object_new() is not enough?

`GstGhostPad` is one I remember currently. Basically everything where the `foo_new()` functions do more than just calling `g_object_new()`.

> You're right, I forgot the properties without default values. For that, I thought I could just add the required properties in the new() method.

All properties have default values (unless they're object/boxed types, in which case the default is always `None`). But those default values might not be enough for constructing a valid, usable object.